File: 128. Longest Consecutive Sequence/longestConsecutive.py
from typing import List

class Solution:
  def longestConsecutive(self, nums: List[int]) -> int:
    numSet = set(nums)
    longest = 0

    for n in nums:
      if (n - 1) in numSet:
        continue
      length = 0
      while n in numSet:
        n += 1
        length += 1
      longest = max(longest, length)

    return longest


solution = Solution()
nums = [100, 4, 200, 1, 3, 2]
print('longestConsecutive: ', solution.longestConsecutive(nums))

File: 128. Longest Consecutive Sequence/longestConsecutive.js
/**
 * @param {number[]} nums
 * @return {number}
 */
const longestConsecutive = (nums) => {
  const seen = new Set(nums);
  let longest = 0;

  for (let i = 0; i < nums.length; i++) {
    if (seen.has(nums[i] - 1)) {
      continue;
    }

    let len = 0;
    while (seen.has(nums[i] + len)) {
      len++;
    }
    longest = Math.max(longest, len);
  }

  return longest;
};


const nums = [100, 4, 200, 1, 3, 2];
console.log('longestConsecutive: ', longestConsecutive(nums));
